tree.go

```diff
@@ -7,30 +7,25 @@ import (
 	"github.com/NebulousLabs/merkletree"
 )
 
-var DefaultTreeCreator = &defaultTreeCreator{}
-
 type Tree interface {
 	Push(data []byte)
 	// TODO(ismail): is this general enough?
 	Prove(idx int) (merkleRoot []byte, proofSet [][]byte, proofIndex uint64, numLeaves uint64)
 	Root() []byte
 }
 
-type TreeCreator interface {
-	New() Tree
-}
-
-var _ TreeCreator = &defaultTreeCreator{}
 var _ Tree = &DefaultTree{}
```

Comment:
How would developers use their own tree implementation for the library? Create a struct that implements Tree, then supply a function that creates a new tree?

Reply:
> Create a struct that implements Tree, then supply a function that creates a new tree?

Yes, exactly.